src/documentation/execution-model.md

```diff
@@ -0,0 +1,163 @@
+---
+layout: default
+title: "Beam Execution Model"
+permalink: /documentation/execution-model/
+---
+
+# Apache Beam Execution Model
+
+The Beam model allows runners to execute your pipeline in different ways.
+Depending on the runner’s choices, you may observe various effects as a result.
+This page describes the effects of these choices so you can better understand
+how Beam pipelines execute.
+
+**Table of Contents:**
+* TOC
+{:toc}
+
+## Processing of elements
+
+The serialization and communication of elements between machines is one of the
+most expensive operations in a distributed execution of your pipeline. Avoiding
+this serialization may require re-processing elements after failures or may
+limit the distribution of output to other machines.
+
+The runner processes elements on many machines and may serialize elements in
+between machines for other communication and persistence reasons.
```

Comment:
This sentence is a little odd. Maybe "for communication and other reasons, such as persistence"?

Reply:
Or is this redundant with the first sentence of the next section?

Reply:
yeah, redundant... I did some rewriting of this sentence, ptal